crates/client/src/client.rs

```diff
@@ -874,7 +887,8 @@ impl FuelClient {
         &self,
         request: PaginationRequest<String>,
     ) -> io::Result<PaginatedResult<types::Block, String>> {
-        let query = schema::block::BlocksQuery::build(request.into());
+        let query =
+            schema::block::BlocksQuery::build(schema::ConnectionArgs::from(request));
```

Comment:
nit: There are a bunch of spots now with these nested function calls. It would be much easier to read as:
```suggestion
            let args = schema::ConnectionArgs::from(request);
            schema::block::BlocksQuery::build(args);
```
I won't mark all the places, but there a like 6+ places.